client/my-sites/theme/main.jsx

```diff
@@ -1303,8 +1335,14 @@ class ThemeSheet extends Component {
 			plansUrl = localizeUrl( 'https://wordpress.com/pricing' );
 		} else if ( siteSlug ) {
 			const redirectTo = `/theme/${ themeId }${ section ? '/' + section : '' }/${ siteSlug }`;
-			const plan = isExternallyManagedTheme || isBundledSoftwareSet ? PLAN_BUSINESS : PLAN_PREMIUM;
+			let plan;
+			if ( ! isThemeAllowedOnSite ) {
+				plan = THEME_TIERS[ themeTier.slug ].minimumUpsellPlan;
+			} else {
+				plan = isExternallyManagedTheme || isBundledSoftwareSet ? PLAN_BUSINESS : PLAN_PREMIUM;
+			}
 
+			// @TODO we should add a new feature for personal plan themes or what we agree on.
```

Comment:
on the back-end we have `WPCOM_Features::PERSONAL_THEMES` so I guess here we should have FEATURE_PERSONAL_THEMES

Reply:
Not sure, but this isn't needed right now, it's just something that we need to keep in mind. Maybe we don't want to have a feature for every theme tier or maybe we do.